content/blog/2025-04-02-valkey-8-1-0-ga.md

```diff
@@ -0,0 +1,212 @@
+
+You can learn more about the design and results in [the dedicated blog post about the implementation](/blog/new-hash-table). 
+
+### Iterator Prefetching
+
+Iterating over the key set keys is done in various scenarios, for example when a Valkey node needs to send all the keys and values to a newly connected replica.
+
+In Valkey 8.1 the iteration functionality is improved by using memory prefetching techniques.
+
+This means that when an element is going to be returned to the caller, the bucket and its elements have already been loaded into CPU cache when the previous bucket was being iterated.
+
+This makes the iterator [3.5x](https://github.com/valkey-io/valkey-io.github.io/pull/227) faster than without prefetching, thus reducing the time it takes to send the data to a newly connected replica.
+
+Commands like `KEYS` and also benefit from this optimization.
+
+### I/O Threads Improvements
+
+Following up the I/O threads improvements added in 8.0, more operations have been offloaded to the I/O thread pool in the 8.1 release, improving the throughput and latency of some operations.
+
+In the new release, TLS connections are now able to offload the TLS negotiation to I/O threads. This change improves the rate of accepting new connections by around [300%](https://github.com/valkey-io/valkey/pull/1338).
+
+Other sources of overhead in the TLS connection handling were identified, namely in the calls to `SSL_pending()` and `ERR_clear_error()` functions, which were being called in the main event thread. By offloading these functions to the I/O threads pool, a throughput improvement was achieved in some operations. For instance, it was observed a [10%](https://github.com/valkey-io/valkey/pull/1271) improvement in `SET` operations throughput, and a [22%](https://github.com/valkey-io/valkey/pull/1271) improvement in `GET` operations throughput.
+
+Replication traffic efficiency was also improved in 8.1 by offloading the reading of replication stream on the replicas to the I/O thread pool which means they can serve more read traffic. On the primaries, replication stream writes are now offloaded to the I/O thread pool.
+
+### Replication Improvements
+
+Full syncs with TLS enabled are up to [18%](https://github.com/valkey-io/valkey/pull/1479) faster by removing redundant CRC checksumming when using diskless replication.
+
+The fork copy-on-write memory overhead is reduced by up to [47%](https://github.com/valkey-io/valkey/pull/905)
+ 
+### Sorted set and hyperloglog and bitcount optimizations
+
+`ZRANK` command, which serves a popular usecase in operating Leaderboards, was optimized to perform up to [45%](https://github.com/valkey-io/valkey/pull/1389) faster, depending on the sorted set size.
```

Comment:
We should mention that we use SIMD to parse floats using fast_float and this needs to be enabled when compiling. We didn't use it by default because it's some new compile-time tooling. It's C++.

This PR: https://github.com/valkey-io/valkey/pull/1260